device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -732,12 +739,90 @@ public void onLinkFlow(Event event)
         logger.LogDebug("Entered in method %s", logger.getMethodName());
 
         // Codes_SRS_AMQPSIOTHUBCONNECTION_15_040: [The event handler shall save the remaining link credit.]
-        this.linkCredit = event.getLink().getCredit();
-        logger.LogDebug("The link credit value is %s, method name is %s", this.linkCredit, logger.getMethodName());
+        this.amqpsSessionManager.onLinkFlow(event);
+
+        processMessagesToSend();
 
         logger.LogDebug("Exited from method %s", logger.getMethodName());
     }
 
+    private void processMessagesToSend()
+    {
+        int messagesAttemptedToBeProcessed = 0;
+        int lastDeliveryTag = 0;
+        com.microsoft.azure.sdk.iot.device.Message message = messagesToSend.poll();
+        while (message != null && messagesAttemptedToBeProcessed < 1000 && lastDeliveryTag >= 0)
+        {
+            messagesAttemptedToBeProcessed++;
```

Comment:
>messagesAttemptedToBeProcessed [](start = 12, length = 30)

reset messagesAttemptedToBeProcessed on subsequent messages polled from the queue?